src/smc-webapp/app-framework/TypedMap.ts

```diff
@@ -43,8 +43,8 @@ export interface TypedMap<TProps extends Record<string, any>>
     ...collections: Array<Partial<TProps> | Iterable<[string, any]>>
   ): this;
 
-  mergeWith(
-    merger: (oldVal: any, newVal: any, key: keyof TProps) => any,
+  mergeWith<K extends keyof TProps>(
+    merger: (oldVal: any, newVal: any, key: K) => any,
```

Comment:
Hopefully my [question on stack overflow](https://stackoverflow.com/questions/58846817/whats-the-difference-between-these-two-key-type-argument-declarations) will clarify why this is necessary...